File: clojure/2.clj
(defn fib [n]
	(if (= n 0)
		1
		(if (= n 1)
			2
			(+ (fib (- n 2)) (fib (- n 1))))))

(defn to-add [n]
	(if (= 0 (rem n 2))
		n
		0))

(defn solve [total current end]
	(if (= current end)
		total
		(recur (+ total (to-add(fib current))) (+ 1 current) end)))

(println (solve 0 0 32))

File: 2.rb
def fib(n)
	if(n == 0)
		1
	elsif(n == 1)
		2
	else
		fib(n-2) + fib(n-1)
	end
end

def is_even(n)
	if(n % 2 == 0)
		true
	else
		false
	end
end

def solve
	total = 0
	(1..31).each do |n|
		f = fib(n)
		if(is_even(f))
			total += f
		end
	end
	puts total
end
